File: 2540-minimum-common-value/Solution.java
// Return the minimum integer common to both arrays

class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        for (int n : nums1) {
            if (binarySearch(nums2, n) >= 0) { return n; }
        }

        return -1;
    }

    public int binarySearch(int[] arr, int x) {
        int s = 0;
        int m = 0;
        int e = arr.length - 1;
        while (s <= e) {
            m = (s + e) / 2;

            if (arr[m] == x) { return m; } 
            else if (arr[m] < x) { s = m + 1; } 
            else { e = m - 1; }
        }

        return -1;
    }
}

File: 2540-minimum-common-value/Solution.js
// Return the minimum integer common to both arrays

/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */

var getCommon = function (nums1, nums2) {
    for (let x of nums1) {
        if (x < nums2[0] || x > nums2[nums2.length - 1]) { continue; }
        if (binarySearch(nums2, x) >= 0) { return x; }
    }

    return -1;
};

function binarySearch(arr, x) {
    let s = 0;
    let m = 0;
    let e = arr.length - 1;
    while (s <= e) {
        m = parseInt((s + e) / 2);

        if (arr[m] == x) { return m; }
        else if (arr[m] < x) { s = m + 1; }
        else { e = m - 1; }
    }

    return -1;
}
